src/block/Note.php

```diff
@@ -69,5 +77,37 @@ public function setPitch(int $pitch) : void{
 		$this->pitch = $pitch;
 	}
 
-	//TODO
+	public function onInteract(Item $item, int $face, Vector3 $clickVector, ?Player $player = null) : bool{
+		if($this->getSide(Facing::UP)->getId() !== BlockLegacyIds::AIR){
+			return false;
+		}
+		$this->pitch = ++$this->pitch % (self::MAX_PITCH + 1);
+		$this->pos->getWorld()->setBlock($this->pos, $this);
+		$this->triggerNote();
+		return true;
+	}
+
+	public function onAttack(Item $item, int $face, ?Player $player = null) : bool{
+		if($this->getSide(Facing::UP)->getId() !== BlockLegacyIds::AIR){
+			return false;
+		}
+		$this->triggerNote();
+		return true;
+	}
+
+	/**
+	 * Get the instrument that the noteblock will play when it is triggered
+	 *
+	 * @return NoteInstrument
+	 */
+	public function getInstrument() : NoteInstrument{
+		$blockUnder = $this->getSide(Facing::DOWN);
+		return $blockUnder->getNoteblockInstrument();
```

Comment:
Are you sure about this? What about when two note blocks are stacked on top of each other?

Reply:
My bad, I misread the function name. Ignore me.

Reply:
It might be a good idea to store this as a property and have it updated in `readStateFromWorld()` and `onNearbyBlockChange()` (better performance).